api/db/db_models.py

```
@@ -1984,6 +1984,108 @@ def alter_db_column_type(migrator, table_name, column_name, new_column_type):
+        return
+
+    if col_type not in _STRING_COLUMN_TYPES:
+        logging.warning("Skipping tenant_model.model_type conversion from unexpected type %s", col_type)
+        return
+
+    try:
+        if settings.DATABASE_TYPE.upper() == "POSTGRES" or is_gaussdb_compatible_database():
+            DB.execute_sql(f'ALTER TABLE "{table_name}" ALTER COLUMN "{column_name}" TYPE integer USING {_tenant_model_type_postgres_using_sql()}')
+        else:
+            name_cases = " ".join(f"WHEN '{name}' THEN '{value}'" for name, value in TENANT_MODEL_TYPE_NAME_TO_INT.items())
+            DB.execute_sql(f"UPDATE `{table_name}` SET `{column_name}` = CASE LOWER(`{column_name}`) {name_cases} ELSE `{column_name}` END")
```

Ruff (0.16.2)

[error] Possible SQL injection vector through string-based query construction

(S608)